### PR TITLE
fix: add swagger docs

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Supersedes #230  
* Adds swagger spec for gotrue's API
* Had to change the response data structure from a `map[string]interface{}` to a proper struct for some endpoint so they can be reused in `/docs` 

## To run
```
# Make sure you install go-swagger locally
$ git clone https://github.com/go-swagger/go-swagger && cd go-swagger

# go-swagger >0.30.0 is broken and does not generate the spec from a struct properly. v0.30.0 is the latest working version.
$ git checkout v0.30.0

# Install the swagger command 
$ go install -ldflags "-X github.com/go-swagger/go-swagger/cmd/swagger/commands.Version=$(git describe --tags) -X github.com/go-swagger/go-swagger/cmd/swagger/commands.Commit=$(git rev-parse HEAD)" ./cmd/swagger

# cd into your local gotrue clone and generate the spec
$ swagger generate spec -o ./swagger.json

# Serve the swagger ui locally
$ swagger serve -F=swagger swagger.json
```

### CODE REVIEW

**Comment by hf** [comment]
Can we add this generation to the CI, so that people never forget to update it locally?

File: main.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"log"
 
 	"github.com/netlify/gotrue/cmd"
+	_ "github.com/netlify/gotrue/docs"
```
Comment (hf) [inline main.go:7]: This is also concerning a bit... is this really necessary? Can't we put a `main` function inside `docs` that does nothing -- if that's important to the tool.

**Comment by hf** [comment]
I'm not sure the generated spec is correct -- if you add it to editor.swagger.io you get this :/ 

image

**Review by J0 — APPROVED** [review]

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          output: swagger-ui
+          spec-file: swagger.json
+
+        - name: Deploy to GitHub Pages
+          uses: peaceiris/actions-gh-pages@v3
+          with:
+            github_token: ${{ secrets.GITHUB_TOKEN }}
+            publish_dir: swagger-ui
+
```
Comment (hf) [inline .github/workflows/docs.yml:28]:
```suggestion
```

**Pull request merged**